### PR TITLE
Update yolov8 example to use Keras Core

### PR DESCRIPTION
**Pull request by IMvision12**
@ianstenbit @fchollet

### CODE REVIEW

File: examples/vision/yolov8.py
Comment (ianstenbit) [inline examples/vision/yolov8.py:117]: Was something wrong with loading the data from Kaggle?

**Review by ianstenbit — COMMENTED** [review]
Thanks for the update! Generally looks good to me, just want to understand what we're doing in terms of data loading.

File: examples/vision/yolov8.py
Comment (IMvision12) [reply examples/vision/yolov8.py:117]: Yes, there was issue with pycocotools...

**Comment by IMvision12** [comment]
@ianstenbit  Also, given script doesn't work with jax backend ...

**Comment by IMvision12** [comment]
@ianstenbit Any updates?

File: examples/vision/yolov8.py
Comment (ianstenbit) [reply examples/vision/yolov8.py:117]: I'd like to avoid loading this from Drive if we can.

Can you share your Kaggle notebook where you were seeing this error? It's probably a simple fix with the dataloading, and it'd be a bit nicer than loading a Drive file where the auth + verification is not as easy.

**Comment by ianstenbit** [comment]
> @ianstenbit Any updates?

Can you elaborate on the issues you were seeing with the JAX backend?

**Comment by IMvision12** [comment]
> Can you elaborate on the issues you were seeing with the JAX backend?

Getting error while using augmentation layers

```python
    ValueError: Only input tensors may be passed as positional arguments. The following argument value should be passed as a keyword argument: tf.RaggedTensor(values=tf.RaggedTensor(values=Tensor("RaggedFromVariant/RaggedTensorFromVariant:2", shape=(None,), dtype=float32), row_splits=Tensor("RaggedFromVariant/RaggedTensorFromVariant:1", shape=(None,), dtype=int64)), row_splits=Tensor("RaggedFromVariant/RaggedTensorFromVariant:0", shape=(5,), dtype=int64)) (of type <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'>)
```

colab : https://colab.research.google.com/drive/1qtAk8GRF1Nt_9TtkkLMPBkqUUiBA3bB2?usp=sharing

**Comment by ianstenbit** [comment]
> > Can you elaborate on the issues you were seeing with the JAX backend?
> 
> Getting error while using augmentation layers
> 
> ```python
>     ValueError: Only input tensors may be passed as positional arguments. The following argument value should be passed as a keyword argument: tf.RaggedTensor(values=tf.RaggedTensor(values=Tensor("RaggedFromVariant/RaggedTensorFromVariant:2", shape=(None,), dtype=float32), row_splits=Tensor("RaggedFromVariant/RaggedTensorFromVariant:1", shape=(None,), dtype=int64)), row_splits=Tensor("RaggedFromVariant/RaggedTensorFromVariant:0", shape=(5,), dtype=int64)) (of type <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'>)
> ```
> 
> colab : https://colab.research.google.com/drive/1qtAk8GRF1Nt_9TtkkLMPBkqUUiBA3bB2?usp=sharing

Ahh I see the issue.
For preprocessing, using `keras.Sequential` is problematic here because that's a Keras Core model that expects JAX inputs.

Instead, try using a `keras_cv.layers.Augmenter` (same API, just swap the symbol)

**Comment by IMvision12** [comment]
@ianstenbit now its working with `keras_cv.layers.Augmenter` and also its working on kaggle, I will push a commit in some time updating code and docs ..

**Comment by IMvision12** [comment]
@ianstenbit Now you can review it ...

File: examples/vision/yolov8.py
Comment (ianstenbit) [inline examples/vision/yolov8.py:6]: nit: s/jax/JAX

**Review by ianstenbit — APPROVED** [review]
Thanks! Looks good to me.

**Comment by ianstenbit** [comment]
I think @jbischof has approval powers here -- we'll also need to render the changes to the guide @IMvision12 (to update the MD and ipynb files)

**Comment by IMvision12** [comment]
Once @jbischof and @fchollet approve these changes then I can add the generated files

**Comment by IMvision12** [comment]
@jbischof Any updates?

**Comment by Paryavi** [comment]
@IMvision12 
I am testing your Yolo colab code, your keras-core version; https://colab.research.google.com/drive/1qtAk8GRF1Nt_9TtkkLMPBkqUUiBA3bB2?usp=sharing
And for training(yolo.fit)
I removed .take(2) in your code;

yolo.fit(
    train_ds,
    steps_per_epoch=2,
    validation_data=val_ds.take(2),
    epochs=3,
    callbacks=[coco_metrics_callback],
)

also, I modified epochs to 100 as follows, but the loss does not improve, 
yolo.fit(
    train_ds,
    steps_per_epoch=2,
    validation_data=val_ds,
    epochs=100,
    callbacks=[coco_metrics_callback],
)
I think I am making a salient mistake! also, do I need to remove /take from the previous line(val_ds.take(2)) in the following;
coco_metrics_callback = keras_cv.callbacks.PyCOCOCallback(
    val_ds.take(2), bounding_box_format="xyxy"
)

Thanks

**Comment by Paryavi** [comment]
Hi @jbischof, can you please review the merge of the above issue?

**Comment by Paryavi** [comment]
@IMvision12 
I ran it again in Colab using T4 machine, it has this issue on Model creation part;
Code:
backbone = keras_cv.models.YOLOV8Backbone.from_preset(
    "yolo_v8_s_backbone_coco"  # We will use yolov8 small backbone with coco weights
)
Error:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-22-370797f3321c>](https://localhost:8080/#) in <cell line: 1>()
----> 1 backbone = keras_cv.models.YOLOV8Backbone.from_preset(
      2     "yolo_v8_s_backbone_coco"  # We will use yolov8 small backbone with coco weights
      3 )

4 frames
[/usr/local/lib/python3.10/dist-packages/keras_cv/models/utils.py](https://localhost:8080/#) in get_tensor_input_name(tensor)
     23         return tensor._keras_history.operation.name
     24     else:
---> 25         return tensor.node.layer.name
     26 
     27 

AttributeError: 'KerasTensor' object has no attribute 'node'
____________________
P.S. : @ianstenbit and @fchollet, could you please take a look at KerasCV? It needs more attention.

**Comment by IMvision12** [comment]
@Paryavi I have opened a issue https://github.com/keras-team/keras-io/pull/813

**Comment by Paryavi** [comment]
@IMvision12 
It seems that issue 813 was resolved.
Can we fix the error I shared in this thread by modifying your YoloV8 example to Keras 3?

**Pull request closed without merging**
